a3p-integration/proposals/a:upgrade-next/provisionPool.test.js

```diff
@@ -0,0 +1,101 @@
+// @ts-check
+
+import test from 'ava';
+
+import {
+  evalBundles,
+  getIncarnation,
+  waitForBlock,
+} from '@agoric/synthetic-chain';
+
+import { bankSend, getProvisionPoolMetrics } from './agd-tools.js';
+
+const NULL_UPGRADE_BANK_DIR = 'upgrade-bank';
+const UPGRADE_PP_DIR = 'upgrade-provisionPool';
+const ADD_STARS_DIR = 'add-STARS';
+const ADD_STARS2_DIR = 'add-STARS2';
+
+const USDC_DENOM =
+  'ibc/295548A78785A1007F232DE286149A6FF512F180AF5657780FC89C009E2C348F';
+const PROVISIONING_POOL_ADDR = 'agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346';
+
+/**
+ * @file
+ * The problems to be addressed are that provisionPool won't correctly handle
+ * (#8722) deposit of assets after it (provisionPool) is upgraded, or (#8724)
+ * new asset kinds after vat-bank is upgraded.
+ *
+ * To test this, we will
+ *
+ * 1. See that we can add USDC.
+ *
+ * 2. Null upgrade vat-bank, and see that we can add a new collateal.
+ *
+ * 2a. if we had been able to null-upgrade provisionPool at this point, we
+ * wouldn't have been able to productively add USDC, but the null upgrade fails.
+ *
+ * 3. Do a full upgrade of provisionPool; then deposit USDC, and see IST
+ * incremented in totalMintedConverted.
+ *
+ * 4. Null upgrade vat-bank again, and then see (in logs) that adding a new
+ * asset type gives the ability to make deposits.
+ */
+
+const contributeToPool = async (t, asset, expectedToGrow) => {
+  const metricsBefore = await getProvisionPoolMetrics();
+  console.log('PPT pre', metricsBefore);
+
+  await bankSend(PROVISIONING_POOL_ADDR, asset);
+
+  const metricsAfter = await getProvisionPoolMetrics();
+  console.log('PPT post', metricsAfter);
+  t.is(
+    metricsAfter.totalMintedConverted.brand,
+    metricsBefore.totalMintedConverted.brand,
+    'brands match',
+  );
+  if (expectedToGrow) {
+    // I couldn't import AmountMath. dunno why.
```

Comment:
you'd have to add `@agoric/ertp` as a dependency of this proposal package. Did you try that?

Reply:
oh yeah, that durn dependency on SES.

You'll also need `import "@endo/init/debug.js"` before that import.

Reply:
I'm not familiar with that one (or I've forgotten). Do you have a pointer to an issue or discussion?

Reply:
Ah, it turns out that what we get from vstorage isn't actually an Amount. The brand field contains stringified captp representations. We can compare them for equality, but they're not the brands that AmountMath recognizes.